packages/MJServer/src/resolvers/GetDataResolver.ts

```diff
@@ -118,12 +118,18 @@ export class GetDataResolver {
                 throw new Error(`Token ${input.Token} is not valid or has expired`);
             }
 
+            // Use the read-only connection for executing queries
+            const readOnlyDataSource = context.dataSources.find(ds => ds.type === 'Read-Only')?.dataSource;
+            if (!readOnlyDataSource) {
+                throw new Error('Read-only data source not found');
+            }
```

Comment:
This is a breaking change since any existing installations that already use this query that have not yet set up a read only DB user will suddenly have their systems fail.